src/test/java/io/managed/services/test/kafka/KafkaInstanceAPITest.java

```diff
@@ -165,6 +172,145 @@ public void testFailToCreateTopicIfItAlreadyExist() {
+                {false, newCE().key("compression.type").value("gzip")},
+
+                {true, newCE().key("file.delete.delay.ms").value("60000")}, // default permitted
+                {false, newCE().key("file.delete.delay.ms").value("1")},
+
+                {true, newCE().key("flush.messages").value(Long.toString(Long.MAX_VALUE))}, // default permitted
+                {false, newCE().key("flush.messages").value("1")},
+
+                {true, newCE().key("flush.ms").value(Long.toString(Long.MAX_VALUE))}, // default permitted
+                {false, newCE().key("flush.ms").value("1")},
+
+                {false, newCE().key("follower.replication.throttled.replicas").value("*")},
+                {false, newCE().key("follower.replication.throttled.replicas").value("1:1")},
+
+                {true, newCE().key("index.interval.bytes").value("4096")}, // default permitted
+                {false, newCE().key("index.interval.bytes").value("1")},
+
+                {false, newCE().key("leader.replication.throttled.replicas").value("*")},
+                {false, newCE().key("leader.replication.throttled.replicas").value("1:1")},
+
+                {true, newCE().key("max.message.bytes").value(Integer.toString(messageSizeLimit))},
+                {true, newCE().key("max.message.bytes").value("1")},
+                {true, newCE().key("max.message.bytes").value(Integer.toString(messageSizeLimit - 1))},
+                {false, newCE().key("max.message.bytes").value(Integer.toString(messageSizeLimit) + 1)},
+
+                {false, newCE().key("message.format.version").value("3.0")},
+                {false, newCE().key("message.format.version").value("2.8")},
+                {false, newCE().key("message.format.version").value("2.1")},
+
+                {true, newCE().key("min.cleanable.dirty.ratio").value("0.5")},
+                {false, newCE().key("min.cleanable.dirty.ratio").value("0")},
+                {false, newCE().key("min.cleanable.dirty.ratio").value("1")},
+
+                //{true, newCE().key("min.insync.replicas").value(desiredBrokerCount > 2 ? "2" : "1")}, // TODO the specification says we should permit 2 for standard and 1 for developer
+                {desiredBrokerCount < 3, newCE().key("min.insync.replicas").value("1")},
+
+                {true, newCE().key("segment.bytes").value(Integer.toString(fiftyMi))},
+                {true, newCE().key("segment.bytes").value(Integer.toString(fiftyMi + 1))},
+                {false, newCE().key("segment.bytes").value(Integer.toString(fiftyMi - 1))},
+                {false, newCE().key("segment.bytes").value(Integer.toString(1))},
+
+                {true, newCE().key("segment.index.bytes").value(Integer.toString(tenMi))},
+                {false, newCE().key("segment.index.bytes").value("1")},
+
+                //{false, newCE().key("segment.jitter.ms").value("0")},  // TODO - the specification says "We should prevent users from setting it", but the code does enforce that.
```

Comment:
@showuon the shows the implementation of the policy disagrees with the epic brief.

Reply:
Oops, it's my bad. I misread the JIRA description. Yes, it should be `immutable`. I'll update it. Thanks for the test and catch it.